spec/models/advocate_spec.rb

```diff
@@ -32,30 +32,30 @@
 
   it { should validate_inclusion_of(:role).in_array(%w( admin advocate )) }
 
-  it { should validate_presence_of(:account_number) }
+  it { should validate_presence_of(:supplier_number) }
 
   context 'account number validation' do
 
     it 'should fail validation if too long' do
-      a = FactoryGirl.build :advocate, account_number: 'ACC123'
+      a = FactoryGirl.build :advocate, supplier_number: 'ACC123'
       expect(a).not_to be_valid
-      expect(a.errors[:account_number]).to eq( ['must be 5 alhpa-numeric characters'] )
+      expect(a.errors[:supplier_number]).to eq( ['must be 5 alhpa-numeric characters'] )
     end
 
     it 'should fail validation if too short' do
-      a = FactoryGirl.build :advocate, account_number: 'AC12'
+      a = FactoryGirl.build :advocate, supplier_number: 'AC12'
       expect(a).not_to be_valid
-      expect(a.errors[:account_number]).to eq( ['must be 5 alhpa-numeric characters'] )
+      expect(a.errors[:supplier_number]).to eq( ['must be 5 alhpa-numeric characters'] )
     end
 
     it 'should fail validation if not alpha-numeric' do
-      a = FactoryGirl.build :advocate, account_number: 'AC-123'
+      a = FactoryGirl.build :advocate, supplier_number: 'AC-123'
```

Comment:
this might fail validation just because it is too long, rather than non-alpha numeric - "AC-23" better?!